src/adapter/src/catalog.rs

```diff
@@ -2570,7 +2570,9 @@ impl CatalogItemRebuilder {
         id: GlobalId,
         ancestor_ids: &BTreeMap<GlobalId, GlobalId>,
     ) -> Self {
-        if id.is_system() && (entry.is_table() || entry.is_introspection_source()) {
+        if id.is_system()
+            && (entry.is_table() || entry.is_introspection_source() || entry.is_source())
+        {
```

Comment:
Needed this to fix the CI legacy upgrade test.
More context: https://materializeinc.slack.com/archives/C0246GEHL8N/p1683647055760979